firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/Power/AlarmsService.cpp

```diff
@@ -9,15 +9,34 @@
 
 namespace Pufferfish::Driver::Power {
 
+void AlarmsService::generate_alarm(
+    const LogEventCode &alarm_code, bool alarm_active, Application::AlarmsManager &alarms_manager) {
+  if (alarm_active) {
+    alarms_manager.activate_alarm(alarm_code, LogEventType::LogEventType_system);
+  } else {
+    alarms_manager.deactivate_alarm(alarm_code);
+  }
+}
+
 // AlarmsService
 void AlarmsService::transform(
-    const MCUPowerStatus &mcu_power_status, Application::AlarmsManager &alarms_manager) {
+    const MCUPowerStatus &mcu_power_status, Application::AlarmsManager &alarms_manager) const {
+  bool charger_disconnected = false;
   if (!mcu_power_status.charging) {
-    alarms_manager.activate_alarm(
-        LogEventCode_charger_disconnected, LogEventType::LogEventType_system);
-  } else {
-    alarms_manager.deactivate_alarm(LogEventCode_charger_disconnected);
+    charger_disconnected = true;
+  }
```

Comment:
Can this be simplified to `bool charger_disconnected = !mcu_power_status.charging`?